gh-122058: `Lib/inspect`: Update docstrings for `isfunction`, `isgenerator`, `isframe`, `iscode`. (#122059)
python/cpython: Doc/library/inspect.rst

|                 | f_trace           | tracing function for this |
|                 |                   | frame, or ``None``        |
+-----------------+-------------------+---------------------------+

> |                 | f_trace_lines     | is a tracing event        |

Reviewer: I think it's better not to use questions and use one of the following formulations:

- indicate whether a tracing event is triggered for each source line
- true if a tracing event is triggered for each source line, false otherwise

|                 |                   | triggered for each source |
|                 |                   | line?                     |
+-----------------+-------------------+---------------------------+

> |                 | f_trace_opcodes   | are per-opcode events     |

Reviewer: - indicate whether per-opcode events are requested
- true if per-opcode events are requested, false otherwise.

|                 | f_trace           | tracing function for this |
|                 |                   | frame, or ``None``        |
+-----------------+-------------------+---------------------------+
|                 | f_trace_lines     | indicate whether a        |
|                 |                   | tracing event is          |
|                 |                   | triggered for each source |
|                 |                   | source line               |
+-----------------+-------------------+---------------------------+
|                 | f_trace_opcodes   | indicate whether          |
|                 |                   | per-opcode events are     |
|                 |                   | requested                 |
+-----------------+-------------------+---------------------------+

> |                 | clear             | used to clear all         |

Reviewer 2: Would it make sense to add parenthesis to methods? Such as: `clear()`.

Author: I somehow lost this during the previous stage of the review...
Yes, that definitely makes things more obvious. Additionally, I will add parentheses to the docstrings.

> |                 | clear             | used to clear all         |
> |                 |                   | references to local       |
> |                 |                   | variables                 |

Reviewer 2: The docstring says *most*, not *all*, references:

```suggestion
|                 | clear             | used to clear most        |
|                 |                   | references held by the    |
|                 |                   | frame                     |
```

Author: I can't find where it's says _most_
From [doc](https://docs.python.org/3.14/reference/datamodel.html#frame.clear):
> This method clears all references to [local variables](https://docs.python.org/3.14/reference/executionmodel.html#naming) held by the frame. Also, if the frame belonged to a [generator](https://docs.python.org/3.14/glossary.html#term-generator), the generator is finalized. This helps break reference cycles involving frame objects (for example when catching an [exception](https://docs.python.org/3.14/library/exceptions.html#bltin-exceptions) and storing its [traceback](https://docs.python.org/3.14/reference/datamodel.html#traceback-objects) for later use).

Reviewer 2: I copied the docstring. If the docstring is outdated, would you mind to also updated it?

Author: Will do!

|                 |                   | arguments and local       |
|                 |                   | variables                 |
+-----------------+-------------------+---------------------------+
|                 | co_lines          | returns an iterator that  |
|                 |                   | yields successive         |
|                 |                   | bytecode ranges           |
+-----------------+-------------------+---------------------------+
|                 | co_positions      | returns an iterator of    |
|                 |                   | source code positions for |
|                 |                   | each bytecode instruction |
+-----------------+-------------------+---------------------------+

> |                 | replace           | return a copy of the code |

Reviewer: return -> returns (sorry you'll need to expand the table...)

Author: If I understand you correctly, you mean to expand only this section of the table, right? Not the entire table by adding an extra column character (the diff will be an unexpectedly big for this PR..)

Reviewer: Either:

- You should align the `|` so you need to expand the entire table (I think?)
- Or you just add a new line before "code" (it'll be a bit ugly, but you won't change the table width at least).

|                 |                   | arguments and local       |
|                 |                   | variables                 |
+-----------------+-------------------+---------------------------+
|                 | co_lines          | returns an iterator that  |
|                 |                   | yields successive         |
|                 |                   | bytecode ranges           |
+-----------------+-------------------+---------------------------+
|                 | co_positions      | returns an iterator of    |
|                 |                   | source code positions for |
|                 |                   | each bytecode instruction |
+-----------------+-------------------+---------------------------+
|                 | replace           | returns a copy of the     |
|                 |                   | code object with a new    |

> |                 |                   | values                    |

Reviewer 2: "with a new values"? is that correct in English? I'm asking but I'm not a native English speaker.

Author: I'm also not a native English speaker :(
I think the article 'a' is unnecessary, but it needs confirmation from a native speaker.

Reviewer 2: I would say: "with new values".
